File: Airline Reservation System/backend/src/main/java/com/coforge/ars/service/BookingService.java
package com.coforge.ars.service;

import java.util.List;
import java.util.Optional;

import com.coforge.ars.model.Booking;

public interface BookingService {

	boolean saveBooking(Booking booking);

	boolean updateBooking(Integer bookingId, Booking booking);

	boolean deleteBookingById(Integer bookingId);

	Optional<Booking> findByBookingId(Integer bookingId);

	List<Booking> findAllBookings();
	
	List<Booking> findBookingsByCustomerId(Integer customerId);

}

File: Airline Reservation System/frontend/src/services/BookingService.js
import axios from "axios";

const BASE_URL = "http://localhost:1515/api/v1/bookings";

class BookingService {

    saveBooking(booking) {
        return axios.post(BASE_URL, booking);
    }

    findAllBookings() {
        return axios.get(BASE_URL);
    }

    findBookingById(id) {
        return axios.get(BASE_URL + "/" + id);
    }

    updateBooking(id, booking) {
        return axios.put(BASE_URL + "/" + id, booking);
    }

    deleteBooking(id) {
        return axios.delete(BASE_URL + "/" + id);
    }

    findBookingsByCustomerId(customerId) {
        return axios.get(BASE_URL + "/customer/" + customerId);
    }

    downloadTicket(bookingId) {
        return axios.get(
            BASE_URL + "/" + bookingId + "/ticket",
            {
                responseType: "blob"
            }
        );
    }

}

export default new BookingService();